ptf/saibridgeport.py

```diff
@@ -0,0 +1,381 @@
+# Copyright 2021-present Intel Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Thrift SAI interface Bridge Port tests
+"""
+
+from sai_thrift.sai_headers import *
+
+from ptf.testutils import *
+from ptf.packet import *
+from ptf.thriftutils import *
+from sai_base_test import *
+
+
+class BridgePortAttributeTest(SaiHelper):
+    '''
+    Verifies bridge port attributes getting and setting
+    '''
+
+    def setUp(self):
+        super(BridgePortAttributeTest, self).setUp()
+
+        self.port24_bp = sai_thrift_create_bridge_port(
+            self.client,
+            bridge_id=self.default_1q_bridge,
+            port_id=self.port24,
+            type=SAI_BRIDGE_PORT_TYPE_PORT,
+            admin_state=True,
+            fdb_learning_mode=SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW)
+        self.assertTrue(self.port24_bp != 0)
+
+    def runTest(self):
+        # bridge_id
+        # get
+        attr = sai_thrift_get_bridge_port_attribute(
+            self.client, self.port24_bp, bridge_id=True)
+        self.assertEqual(attr['bridge_id'], self.default_1q_bridge)
+
+        # type
+        # get
+        attr = sai_thrift_get_bridge_port_attribute(
+            self.client, self.port24_bp, type=True)
+        self.assertEqual(attr['type'], SAI_BRIDGE_PORT_TYPE_PORT)
+
+        # port_id
+        # get
+        attr = sai_thrift_get_bridge_port_attribute(
+            self.client, self.port24_bp, port_id=True)
+        self.assertEqual(attr['port_id'], self.port24)
+
+        # fdb_learning_mode
+        # get
+        attr = sai_thrift_get_bridge_port_attribute(
+            self.client, self.port24_bp, fdb_learning_mode=True)
+        self.assertEqual(attr['fdb_learning_mode'],
+                         SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW)
+        # set
+        status = sai_thrift_set_bridge_port_attribute(
```

Comment:
Is it possible to combine both set+get+assert check in one common method? This can be used for all the objects and attribtues.

Reply:
This is possible but IMO would be unclear. We would probably have to pass a function as an argument for this method and/or at least its argument we want to set and get. Not sure if this would be a good approach.